src/dappsExplorer/FavoriteDappsSection.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import { useSelector } from 'react-redux'
+import { favoriteDappsSelector } from 'src/dapps/selectors'
+import { ActiveDapp, DappFilter, DappSection, isDappV2 } from 'src/dapps/types'
+import DappCard from 'src/dappsExplorer/DappCard'
+import { NoResults } from 'src/dappsExplorer/NoResults'
+import StarIllustration from 'src/icons/StarIllustration'
+import Colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+interface Props {
+  onPressDapp: (dapp: ActiveDapp) => void
+  filter: DappFilter | null
+  removeFilter: () => void
+}
+
+export function FavoriteDappsSection({ onPressDapp, filter, removeFilter }: Props) {
+  const { t } = useTranslation()
+  const favoriteDapps = useSelector(favoriteDappsSelector)
+
+  // If no favorites, display no favorites section
+  if (favoriteDapps.length === 0) {
+    return (
+      <View style={styles.container}>
+        <StarIllustration />
+        <View style={styles.contentContainer}>
+          <Text style={styles.title}>{t('dappsScreen.noFavorites.title')}</Text>
+          <Text style={styles.description}>{t('dappsScreen.noFavorites.description')}</Text>
+        </View>
+      </View>
+    )
+  }
+
+  // If dapp filter enabled display favorites matching filter or no results section
+  // Favorites dapps matching category filter
+
+  const favoritedDappsFiltered = favoriteDapps.filter(
+    (dapp) => isDappV2(dapp) && (dapp.categories.includes(filter?.id ?? '') || filter?.id === 'all')
+  )
+
+  // Matching favorite dapp(s) found
+  if (favoriteDapps.length > 0 && favoritedDappsFiltered.length > 0) {
```

Comment:
i think the `favoriteDapps.length > 0` check is redundant since we already return out of this function in line 26 if this is false

Reply:
fixed in d2b815fb337c8617bcf131e8dfec60e69a7f81d4!